unit/java_bytecode/generate_concrete_generic_type/generate_java_generic_type.cpp

```diff
@@ -330,7 +333,7 @@ SCENARIO(
         "table")
       {
         const std::string specialised_string =
-          "<java::array[reference]of_"
+          "<array[reference]of_"
```

Comment:
should the `java::` prefix in the generic type argument disappear, too?

Reply:
No because if it does, the tests fail.

Reply:
That doesn't feel like a terribly satisfactory answer :-) Like @mgudemann  I'd have expected all the `java::` prefixes to need to disappear too - but I assume that making that prefix disappear (if desired) is a different piece of work to this PR?

Reply:
Due to the change to the pretty print function that is being used, the `java::` prefixes are only pruned  if they are in the beginning of a string. 

These `array[reference]of_<something>` are being manifest in the goto functions to my understanding, and not in the tests generated, and at the point of their removal, they are just strings. This means that to remove all `java::` prefixes in a type, you would do it one of the two ways: 

1) either intercept sooner, when it's still a `typet` and recursively convert the type, or
2) some regex magic on the strings, along with some rewriting.

Since these strings are not manifest in the tests we generate, and appear relatively harmless, I think it's outside the scope of this task to investigate exactly how and where to change them.